advanced_message_passing.md

```diff
@@ -1,8 +1,112 @@
+```
+
+Example URL:
+
+```text
+http://example.com/path?_dc=eyJ0eXBlIjoiaHR0cHM6Ly9kaWRjb21tLm9yZy9teXByb3RvY29sLzAuMS9teW1lc3NhZ2V0eXBlIiwiaWQiOiI2OTIxMmEzYS1kMDY4LTRmOWQtYTJkZC00NzQxYmNhODlhZjMiLCJmcm9tIjoiZGlkOmV4YW1wbGU6YWxpY2UiLCJib2R5Ijp7Im15X2F0dHJpYnV0ZSI6ICJteXZhbHVlIn19
+```
+
+DIDComm message URLs can be transferred via any method that can send text, including an email, SMS, posting on a website, or QR Code.
+
+Example Email Message:
+
+```email
+To: alice@alum.faber.edu
+From: studentrecords@faber.edu
+Subject: Your request to connect and receive your graduate verifiable credential
+
+Dear Alice,
+
+To receive your Faber College graduation certificate, click here to [connect](http://example.com/path?_dc=eyJ0eXBlIjoiaHR0cHM6Ly9kaWRjb21tLm9yZy9teXByb3RvY29sLzAuMS9teW1lc3NhZ2V0eXBlIiwiaWQiOiI2OTIxMmEzYS1kMDY4LTRmOWQtYTJkZC00NzQxYmNhODlhZjMiLCJmcm9tIjoiZGlkOmV4YW1wbGU6YWxpY2UiLCJib2R5Ijp7Im15X2F0dHJpYnV0ZSI6ICJteXZhbHVlIn19) with us, or paste the following into your browser:
+
+http://example.com/path?_dc=eyJ0eXBlIjoiaHR0cHM6Ly9kaWRjb21tLm9yZy9teXByb3RvY29sLzAuMS9teW1lc3NhZ2V0eXBlIiwiaWQiOiI2OTIxMmEzYS1kMDY4LTRmOWQtYTJkZC00NzQxYmNhODlhZjMiLCJmcm9tIjoiZGlkOmV4YW1wbGU6YWxpY2UiLCJib2R5Ijp7Im15X2F0dHJpYnV0ZSI6ICJteXZhbHVlIn19
+
+If you don't have an identity agent for holding credentials, you will be given instructions on how you can get one.
+
+Thanks,
+
+Faber College
+Knowledge is Good
+```
+
+Example URL encoded as a QR Code:
+
+![Example QR Code](ExampleQRCode.png)
+
+##### Short URL Message Retrieval
+
+It seems inevitable that the length of some DIDComm messages will be too long to produce a useable QR code. Techniques to avoid unusable QR codes have been presented above, including using attachment links for requests, minimizing the routing of the response and eliminating unnecessary whitespace in the JSON. However, at some point a _sender_ may need generate a very long URL. In that case, a short URL message retrieval redirection should be implemented by the sender as follows:
+
+- The sender should generate and track a GUID for the out-of-band message URL.
+- The shortened version should be:
+  - `https://example.com/path?_dcid=5f0e3ffb-3f92-4648-9868-0d6f8889e6f3`
+  - Note the replacement of the query parameter `_dc` with `_dcid` when using shortened URL.
+- On receipt of this form of message, the agent must do an HTTP GET to retrieve the associated encoded  message.
+  - A sender may want to wait to generate the full invitation until the redirection event of the shortened URL to the full length form dynamic, so a single QR code can be used for distinct messages.
+
+A usable QR code will always be able to be generated from the shortened form of the URL.
+
+Note: Due to the privacy implications, a standard URL shortening service SHOULD NOT be used.
```

Comment:
This is excellent, but perhaps a reference to some privacy considerations section where an explanation of why can be provided would be helpful.

Reply:
Section added at the top.

Reply:
Added a suggestion in there to help further clarify why they shouldn't be used. Hopefully it helps a bit.